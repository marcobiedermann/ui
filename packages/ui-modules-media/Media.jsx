import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import styles from './media.css';

const Media = ({ children }) => (
  <div className={styles.media}>
    {children}
  </div>
);

Media.propTypes = {
  children: PropTypes.node,
};

Media.defaultProps = {
  children: null,
};

const MediaObject = ({ children, modifier }) => (
  <div
    className={classNames(styles.media__object, {
      [styles[`media__object--${modifier}`]]: modifier,
    })}
  >
    {children}
  </div>
);

MediaObject.propTypes = {
  children: PropTypes.node,
  modifier: PropTypes.string,
};

MediaObject.defaultProps = {
  children: null,
  modifier: 'left',
};

const MediaBody = ({ children }) => (
  <div className={styles.media__body}>
    {children}
  </div>
);

MediaBody.propTypes = {
  children: PropTypes.node,
};

MediaBody.defaultProps = {
  children: null,
};

export default Media;
export {
  MediaObject,
  MediaBody,
};
