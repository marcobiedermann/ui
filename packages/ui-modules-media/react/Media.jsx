import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import styles from '../media.css';

class Media extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
  };

  static defaultProps = {
    children: null,
  };

  render() {
    const { children } = this.props;

    return <div className={styles.media}>{children}</div>;
  }
}

export default Media;
