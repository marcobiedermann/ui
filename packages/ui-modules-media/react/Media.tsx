import React, { FC, HTMLAttributes } from 'react';
import classNames from 'classnames';
import styles from '../media.css';

export interface MediaProps extends HTMLAttributes<HTMLDivElement> {}

const Media: FC<MediaProps> = (props) => {
  const { className, ...otherProps } = props;

  return <div className={classNames(className, styles.media)} {...otherProps} />;
};

export default Media;
