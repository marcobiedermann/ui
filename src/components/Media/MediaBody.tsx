import classNames from 'classnames';
import React, { ReactNode } from 'react';
import styles from '../media.css';

interface MediaBodyProps {
  children: ReactNode;
  className?: string;
}

function MediaBody(props: MediaBodyProps): JSX.Element {
  const { className, ...otherProps } = props;

  return <div className={classNames(className, styles.media__body)} {...otherProps} />;
}

export type { MediaBodyProps };
export default MediaBody;
