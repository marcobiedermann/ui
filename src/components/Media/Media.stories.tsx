import { Meta, Story } from '@storybook/react';
import React from 'react';
import Media from '.';

export default {
  component: Media,
  title: 'Media/Media',
} as Meta;

const Template: Story = (args) => (
  <Media>
    <Media.Object direction={args.direction}>
      <img src="https://via.placeholder.com/48x48" alt="Media Object" />
    </Media.Object>
    <Media.Body>{args.children}</Media.Body>
  </Media>
);

export const Default = Template.bind({});

Default.args = {
  direction: 'left',
  children:
    'Lorem ipsum dolor sit amet consectetur adipisicing elit. Eius officiis ipsam quo omnis dolore quidem esse dignissimos eum laborum adipisci, consequuntur tenetur excepturi labore, ullam animi ut saepe reiciendis ad?',
};
