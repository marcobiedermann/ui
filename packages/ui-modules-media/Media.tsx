import classNames from 'classnames';
import React, { ReactNode } from 'react';
import styles from '../media.css';
import MediaBody from './MediaBody';
import MediaObject from './MediaObject';

interface MediaProps {
  children: ReactNode;
  className?: string;
}

function Media(props: MediaProps): JSX.Element {
  const { className, ...otherProps } = props;

  return <div className={classNames(className, styles.media)} {...otherProps} />;
}

Media.Body = MediaBody;
Media.Object = MediaObject;

export type { MediaProps };
export default Media;
