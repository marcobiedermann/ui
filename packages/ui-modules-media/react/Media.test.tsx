import React from 'react';
import renderer from 'react-test-renderer';
import Media from '.';

const { Body, Object } = Media;

describe('Media component', () => {
  it('renders left correctly', () => {
    const tree = renderer
      .create(
        <Media>
          <Object>Media Object</Object>
          <Body>Media Body</Body>
        </Media>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders right correctly', () => {
    const tree = renderer
      .create(
        <Media>
          <Object direction="right">Media Object</Object>
          <Body>Media Body</Body>
        </Media>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
