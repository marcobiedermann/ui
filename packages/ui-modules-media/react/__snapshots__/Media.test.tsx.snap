// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media component renders left correctly 1`] = `
<div
  className="media"
>
  <div
    className="media__object media__object--left"
  >
    Media Object
  </div>
  <div
    className="media__body"
  >
    Media Body
  </div>
</div>
`;

exports[`Media component renders right correctly 1`] = `
<div
  className="media"
>
  <div
    className="media__object media__object--right"
  >
    Media Object
  </div>
  <div
    className="media__body"
  >
    Media Body
  </div>
</div>
`;
